timeseries/src/autogluon/timeseries/predictor.py

```diff
@@ -61,6 +61,9 @@ class TimeSeriesPredictor:
         - ``"RMSE"``: root mean squared error
 
         For more information about these metrics, see https://docs.aws.amazon.com/forecast/latest/dg/metrics.html.
+    eval_metric_seasonal_period : int, default = 1
```

Comment:
Can we create some logic to calculate the seasonal_error based on the freq? Some example to follow in GluonTS:https://github.com/awslabs/gluonts/blob/176c533dedcbfdf5616021abba277b93304405c8/src/gluonts/evaluation/metrics.py#LL21C16

Reply:
Agree that different framework implement the seasonality differently. We use GluonTS models internally so wondering if we are overriding this value in any way,  otherwise would it introduce discrepancy when we evaluate the model performance

Reply:
Caner ran a comparison a while back and found that for local models that we implement/wrap ourselves (ETS, Theta, SeasonalNaive) setting `seasonal_period=7` on `freq="D"` led to better MAPE/wQL on some benchmark datasets compared to the default GluonTS seasonality mapping of `seasonal_period=1`. 

Also, as far as I know, none of the GluonTS models that we use (DeepAR, TFT, SFF) depend on seasonality, and we don't rely on the metrics computed by GluonTS in any way.

Back to your original question
- I would suggest to set `eval_metric_seasonal_period=None` by default and doing the inference automatically based on `freq`. This is the right thing to do and hopefully this won't affect the users negatively.
- I would prefer to keep [our freq->seasonal_period mapping](https://github.com/autogluon/autogluon/blob/master/timeseries/src/autogluon/timeseries/utils/seasonality.py#L5) as-is instead of changing to the GluonTS version.  

Does this sound reasonable to you?

Reply:
Done 👍